domain/src/main/java/org/oppia/android/domain/translation/TranslationController.kt

```diff
@@ -120,9 +150,24 @@ class TranslationController @Inject constructor(
    *     result is ever provided)
    */
   fun updateAppLanguage(profileId: ProfileId, selection: AppLanguageSelection): DataProvider<Any> {
-    return dataProviders.createInMemoryDataProviderAsync(UPDATE_APP_LANGUAGE_DATA_PROVIDER_ID) {
+    val deferred = retrieveLanguageContentCacheStore(profileId).storeDataAsync(
+      updateInMemoryCache = true
+    ) {
+      AppLanguageSelection.newBuilder().apply {
```

Comment:
We conventionally convert the provided value to a builder and overwrite only those values which are changing, e.g.:

```kotlin
it.toBuilder().apply { selectedLanguage = selection.selectedLanguage }.build()
```

Reply:
Updated.